cluster/saltbase/salt/kube-addons/init.sls

```diff
@@ -151,14 +151,14 @@ addon-dir-create:
 {% endif %}
 
 {% if pillar.get('enable_node_problem_detector', '').lower() == 'true' %}
-/etc/kubernetes/addons/node-problem-detector/node-problem-detector.yaml:
-  file.managed:
-    - source: salt://kube-addons/node-problem-detector/node-problem-detector.yaml.in
-    - template: jinja
```

Comment:
all you need to do is remove this line and remove the .in from the source. This doesn't need recurse.